### PR TITLE
fix: Make all public views available without login

### PR DESCRIPTION
**Pull request by Artur-**
Previously only a root mapped ("") view was available

Fixes https://github.com/vaadin/flow/issues/11355

### CODE REVIEW

**Comment by pleku** [comment]
```
RequestUtilTest.testAnonymousRouteRequest_fooMappedServlet_publicView
 ...
 	at com.vaadin.flow.spring.security.RequestUtilTest.testAnonymousRouteRequest_fooMappedServlet_publicView(RequestUtilTest.java:193)
```
```
RequestUtilTest.testAnonymousRouteRequest_rootMappedServlet_publicView
...
at com.vaadin.flow.spring.security.RequestUtilTest.testAnonymousRouteRequest_rootMappedServlet_publicView(RequestUtilTest.java:160)
```

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java [hunk 1]
```diff
@@ -109,6 +109,11 @@ public boolean isAnonymousEndpoint(HttpServletRequest request) {
     public boolean isAnonymousRoute(HttpServletRequest request) {
         String vaadinMapping = configurationProperties.getUrlMapping();
         String requestedPath = getRequestPathInsideContext(request);
+        if (requestedPath.startsWith("/")) {
```
Comment (pleku) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java:112]: What is this public API that has no javadocs ? Please fix that too.

**Review by pleku — CHANGES_REQUESTED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java (hunk 1 above)
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/RequestUtil.java]: Fixed

**Comment by Artur-** [comment]
Fixed tests and added a few more

**Review by pleku — APPROVED** [review]

**Pull request merged**